### PR TITLE
NE-1286: Python tests, config and formatting

### PR DESCRIPTION
**Pull request by anthonyjpesce**
## Description

This PR:
- Arranges the file naming and directory structure for Python to get closer to our preferred format
- Adds some pre-commit hooks for Python to format and flake8, among other things
- Adds a basic framework for unit testing the Python package
- Runs black and isort over all the files
- Other small tweaks to get things working

## Link to Jira Ticket

[NE-1286]

## Test Steps

Run `pre-commit install`, then `pre-commit run --all-files` to test the pre-commit hooks

Run `tox` to test the Python unit tests

[NE-1286]: https://arcpublishing.atlassian.net/browse/NE-1286?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+from setuptools import setup
+
+# get current directory
+CURRENT_DIR = os.getcwd()
+PARENT_DIR = os.path.dirname(CURRENT_DIR)
+
+fastfec_library = Extension(
+    "fastfec_lib",
+    [
```
Comment (freedmand) [inline python/setup.py:16]: This is so ugly I wonder if we should just map the array with the os join

File: python/src/fastfec/client.py [hunk 2]
```diff
@@ -13,13 +13,23 @@
...
+    provide_line_callback,
+    provide_read_callback,
+    provide_write_callback,
+)
 
 
 class LibFastFEC:
-    """Python wrapper for the fastfec library"""
+    """
+    Python wrapper for the fastfec library
```
Comment (freedmand) [inline python/src/fastfec/client.py:31]: You fixed the ugly block quotes for me. Thanks!!!

File: python/src/fastfec/utils.py [hunk 3]
```diff
@@ -76,44 +87,58 @@ def find_fastfec_lib():
...
 
 def as_bytes(text):
-    """Converts text to bytes, or leaves intact if already bytes or none"""
+    """
+    Converts text to bytes, or leaves intact if already bytes or none
+    """
     if isinstance(text, str):
         # Convert to bytes if in string form
-        return text.encode('utf8')
+        return text.encode("utf8")
```
Comment (freedmand) [inline python/src/fastfec/utils.py:95]: We technically can omit the param since it's a default

Comment (freedmand) [reply python/src/fastfec/utils.py:95]: (I know this is code I wrote)

File: python/src/fastfec/utils.py [hunk 4]
```diff
@@ -184,7 +215,8 @@ def write_callback(filename, extension, contents, num_bytes):
             if not file_descriptor:
                 # Open the file
                 write_cache.file_descriptors[path] = open_function(
-                    path.decode("utf8"), mode='wb')
+                    path.decode("utf8"), mode="wb"
```
Comment (freedmand) [inline python/src/fastfec/utils.py:218]: Same parameter omission for encoding here

File: python/setup.py (hunk 1 above)
Comment (anthonyjpesce) [reply python/setup.py:16]: yeah that's better

File: python/src/fastfec/utils.py (hunk 3 above)
Comment (anthonyjpesce) [reply python/src/fastfec/utils.py:95]: Out of scope for this PR IMO, I just ran the formatter

File: python/tests/test_client.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    to the package's ability to parse summary, contribution and
+    disbursement rows.
+    """
+    with open(filing_1550126, "rb") as f:
+        with FastFEC() as fastfec:
+            parsed = list(fastfec.parse(f))
+
+            # Test the summary data parse
+            summary_form, summary_data = parsed[1]
+            assert len(summary_data.keys()) == 93
```
Comment (freedmand) [inline python/tests/test_client.py:19]: I think you can just `len(summary_data)`

**Review by freedmand — COMMENTED** [review]
So far this is great. Thank you so much for giving this actual Pythonic rigor

File: .gitignore [hunk 6]
```diff
@@ -1,16 +1,37 @@
...
 zig-out
-env/
-build/
-dist/
-MANIFEST
 __pycache__
 *.pyc
+
+# C extensions
+*.so
```
Comment (freedmand) [inline .gitignore:17]: Add `*.dylib` and `*.dll` too

File: python/src/fastfec/__init__.py [hunk 7]
```diff
@@ -0,0 +1,3 @@
+"""Provides the fastfec package"""
```
Comment (freedmand) [inline python/src/fastfec/__init__.py:1]: I think the other `__init__.py` file can be removed (`python/src/__init__.py` — sorry to comment here; GitHub wouldn't let me comment an empty file)

File: python/tests/conftest.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+
+# @pytest.fixture
+# def filing_1527862(get_fixture):
+#     return get_fixture("1527862.fec")
+
+
+# @pytest.fixture
+# def filing_1544132(get_fixture):
+#     return get_fixture("1544132.fec")
```
Comment (freedmand) [inline python/tests/conftest.py:38]: Should these be left commented?

**Review by freedmand — COMMENTED** [review]
Lgtm other than minor comments

**Review by freedmand — APPROVED** [review]

**Pull request merged**